onecodex/viz/_distance.py

```diff
@@ -14,25 +15,26 @@ def _compute_distance(self, rank, metric):
         # if taxonomy trees are inconsistent, unifrac will not work
         if callable(metric):
             distances = metric(self, rank=rank)
-        elif metric in ("braycurtis", "bray-curtis", "bray curtis"):
-            distances = self.beta_diversity(metric="braycurtis", rank=rank)
-        elif metric in ("manhattan", "cityblock"):
-            distances = self.beta_diversity(metric="cityblock", rank=rank)
-        elif metric == "jaccard":
-            distances = self.beta_diversity(metric="jaccard", rank=rank)
-        elif metric in ("unifrac", "weighted_unifrac"):
+        elif metric in (BetaDiversityMetric.BrayCurtis, "bray-curtis", "bray curtis"):
+            distances = self.beta_diversity(metric=BetaDiversityMetric.BrayCurtis, rank=rank)
+        elif metric in ("manhattan", BetaDiversityMetric.CityBlock):
+            distances = self.beta_diversity(metric=BetaDiversityMetric.CityBlock, rank=rank)
+        elif metric == BetaDiversityMetric.Jaccard:
+            distances = self.beta_diversity(metric=BetaDiversityMetric.Jaccard, rank=rank)
+        elif metric == BetaDiversityMetric.WeightedUnifrac:
             distances = self.unifrac(weighted=True, rank=rank)
-        elif metric == "unweighted_unifrac":
+        elif metric == BetaDiversityMetric.UnweightedUnifrac:
```

Comment:
If we expose the enums and recommend their usage in the docs, we can just let people use that and not have to worry about mispelling/other names and they would get auto-completion in notebooks I think?

Reply:
I think our users are probably at the level where enums are pretty far outside their scope of comprehension. I'd prefer to let them just pass in strings for now.

Reply:
Agree with @clausmith.